### ISSUE TITLE
Test Performance of gotbot-opt and do necessary optimizations

### ISSUE THREAD

**Issue by kaushiksk** [IEEE-NITK/DeepNLP#19]
In the gotbot-opt folder run
`
$ python3 gotbot.py --train
`
Might have to train for a long time till perplexity decreases considerably.
Then test performance
`
$ python3 gotbot.py --test-interactive
`

What has been done.

-     Data preprocessing is only done once now by running gotpreprocess.py.
-     While training saved preprocessed data is loaded instead of preprocessing each time.
-     Added a dropout Layer to the Neural Net. Will Hopefully improve performance.
-     Vocabulary consists of the entire text. So during training no \<unknowns\>s will be encountered.

To do

- [ ]     Test if the model improves in performance
- [ ]     Test the model in --test-interactive mode (Haven't done this, might throw unchecked errors)
- [ ]     Run the model on test data set when user interrupts training.
- [ ]     Test performance by removing uncommon words and reducing vocab size. This just requires a small modification in gotpreprocess.py.
- [ ]     Try attention_seq2seq
- [ ]     Try with different embedding size. ( I changed the current one from 32 to 300, this will slow down training but I'm hoping it improves final performance )
